postupgrade/mergeGlobalConfigs.js

```diff
@@ -0,0 +1,174 @@
+const { parse, stringify } = require('comment-json');
+
+/**
+ * @typedef {import('comment-json').CommentJSONObject} CommentJSONObject
+ * @typedef {import('comment-json').CommentToken} CommentToken
+ * 
+ * @typedef {{
+ *   type: 'CommentedOutProperty',
+ *   key: string,
+ *   comment: Omit<CommentToken, 'loc'>
+ * }} CommentedOutPropertyToken
+ * 
+ * @typedef {{
+ *   type: 'Property',
+ *   key: string,
+ *   value: string | number | null
+ *   inlineComment?: Omit<CommentToken, 'loc'>
+ * }} PropertyToken
+ * 
+ * @typedef {PropertyToken | CommentedOutPropertyToken} GlobalConfigToken
+ */
+
+/**
+ * Merges two global configs.
+ * For regular properties, the original config has priority.
+ * For commented out properties, the incoming config has priority.
+ * The `sdkVersion` is a special property which will always favor the incoming config's value,
+ * but the original's inline comment (if one exists).
+ *
+ * This is not intended to handle json with duplicate keys, i.e. a json config
+ * with an `apiKey` property in addition to a commented out `apiKey` property.
+ * 
+ * @param {string} originalConfig 
+ * @param {string} incomingConfig 
+ * @return {string}
+ */
+exports.mergeGlobalConfigs = function mergeGlobalConfigs(originalConfig, incomingConfig) {
+  const originalJson = parse(originalConfig);
+  const incomingJson = parse(incomingConfig);
+  const originalTokens = tokenize(originalJson);
+  const incomingTokens = tokenize(incomingJson);
+  const mergedTokens = mergeTokens(originalTokens, incomingTokens);
+  const commentJsonObject = transformToCommentJsonObject(mergedTokens);
+  if (incomingJson.sdkVersion) {
+    commentJsonObject.sdkVersion = incomingJson.sdkVersion
+  }
+  return stringify(commentJsonObject, null, 2);
+}
+
+/**
+ * Parses a {@link CommentJSONObject} into an array of {@link GlobalConfigToken}s
+ * 
+ * @param {CommentJSONObject}  
+ * @returns {GlobalConfigToken[]}
+ */
+function tokenize(commentJsonObject) {
+  /** @type {GlobalConfigToken[]} */
+  const tokens = []
+
+  /**
+   * @param {CommentToken} comment 
+   * @returns {CommentedOutPropertyToken}
+   */
+  const parseCommentedOutProperty = (comment) => {
+    const propCommentRegex = /^\s?"([^"]+)":.*$/g
+    const val = comment.value
+    const match = [...val.matchAll(propCommentRegex)][0]
+    if (!match || match.length !== 2) {
```

Comment:
In what situations could we hit this conditional? Does being a `CommentToken` not automatically imply that the regex will be matched?

Reply:
the regex matches on strings that looks like a json property, aka a comment that looks like 
```
// "token": "<REPLACE ME>", // The auth token to access Answers experience.
```
Being a comment token just means that it is some kind of comment, it doesn't say anything about what the actual string is like. For instance if we forgot the double quotes around "token". A user could also add their own random comments to the global config, which we've decided not to handle on upgrades.

Reply:
yeah we don't handle arbitrary comments

Reply:
We can also just ignore arbitrary comments and let the merge continue, which I didn't like yesterday but am starting to warm up to